# `foo` cannot be merged with `foo` (at catch.graphql: (4, 5)): they have different `@catch` directives.
query Query1 {
    foo
    foo @catch(to: THROW)
}

# Directive '@catch' cannot be repeated
query Query2 {
    foo @catch(to: THROW) @catch(to: NULL)
}

# 'levels' values must be positive ints
query Query3 {
    foo @catch(levels: [-1], to: THROW)
}

# Invalid 'levels' value '3' for `@catch` usage: this field has a max list level of 2
query Query4 {
    foo @catch(levels: [3], to: THROW)
}

fragment QueryDetails on Query {
    foo @catch(to: THROW)
}

# this is ok because the fragment can have different catch directives
query Query5 {
    ...QueryDetails
    foo @catch(to: NULL)
}

# `foo` cannot be merged with `foo` (at catch.graphql: (36, 9)): they have different `@catch` directives.
query Query6 {
    foo @catch(to: NULL)
    ... on Query {
        foo @catch(to: THROW)
    }
}